lib/contentful.ts

```diff
@@ -41,6 +42,9 @@ export function createClient(params: CreateClientParams): ContentfulClientApi {
     throw new TypeError('Expected parameter space')
   }
 
+  validateResolveLinksParam(params)
```

Comment:
Since chain options are only to be used on the client instance, not directly when creating it, I'm wondering if error messages for these validators are clear enough to be used here too (so users don't try anything like `contentful.withoutLinkResolution.createClient(params)`). What do you think?